server/routerlicious/packages/lambdas/src/alfred/index.ts

```diff
@@ -269,17 +267,25 @@ export function configureWebSocketServices(
 		async function connectDocument(message: IConnect): Promise<IConnectedClient> {
 			const startTime = Date.now();
 			const throttleErrorPerCluster = checkThrottleAndUsage(
-				connectThrottlerPerCluster,
+				throttlersMap?.get("generalCluster")?.get("socketConnections"),
 				getSocketConnectThrottleId("connectDoc"),
 				message.tenantId,
 				logger,
 			);
 			if (throttleErrorPerCluster) {
 				return Promise.reject(throttleErrorPerCluster);
 			}
+
+			const tenantGroup: string | undefined = message.tenantId
+				? tenantThrottlersMap?.get(message.tenantId) ?? undefined
+				: undefined;
 			const throttleErrorPerTenant = checkThrottleAndUsage(
-				connectThrottlerPerTenant,
-				getSocketConnectThrottleId(message.tenantId),
+				tenantGroup
+					? throttlersMap?.get(tenantGroup)?.get("socketConnections")
+					: throttlersMap?.get("generalTenant")?.get("socketConnections"),
+				tenantGroup
+					? `${getSocketConnectThrottleId(message.tenantId)}_${tenantGroup}`
+					: getSocketConnectThrottleId(message.tenantId),
 				message.tenantId,
 				logger,
 			);
```

Comment:
We also have clients per socket connection check on line number 419. Could you also move that check after perTenant throttling? or you can do it as part of your next PR when you move token validation after throttling check for other apis.

`429,
					"Too Many Clients Connected to Document"`

Reply:
I just got a check, and the token validation happened after throttling the check here. :)

Reply:
Got it. I had a try, the clients.length would depend on the call at line 398, which is hard to move that code up.